cmd/version/version.go

```diff
@@ -64,3 +68,62 @@ func displayVersion(cmd *cobra.Command) {
 		Detail:  true,
 	})
 }
+
+func GetLatestVersion() (string, error) {
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error { // no redirect
+			return http.ErrUseLastResponse
+		},
+	}
+	resp, err := client.Get("https://github.com/cisco-open/fsoc/releases/latest")
+	if err != nil {
+		return "", err
+	}
+	split := strings.Split(resp.Header.Get("Location"), "/")
+	if len(split) < 1 {
+		return "", fmt.Errorf("version request did not return a version")
+	}
+	return split[len(split)-1], nil
+}
+
+func CheckForUpdate(versionChannel chan *semver.Version) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			log.Warnf("Error occurred while checking version, continuing")
+		}
+	}()
+	log.Infof("Checking for newer version of FSOC")
+	newestVersion, err := GetLatestVersion()
+	log.Infof("Latest version available: %s", newestVersion)
```

Comment:
```suggestion
	log.Infof("Latest fsoc version available: %s", newestVersion)
```
Remember that the user is not thinking about fsoc itself but about the command they ran. Let's make sure any logged messages are clearly stating what operation they pertain to.

Reply:
Also, please consider using WithField (there is a version for a single field, easier to use than WithFields), so that the field is programmatically parseable in the log